### PR TITLE
Delete failed users when course run grading status is complete

### LINKED ISSUE #3505: Freeze Grades: delete the list of failed users when freezing is COMPLETE
We don't want to keep information about users that failed cache refresh after the course run grading status was set to 'complete'. The freeze task can get triggered for individual learners and make the list of failed users out of date.

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #3505
#### What's this PR do?
Delete failed users when course run grading status is complete

#### How should this be manually tested?
Run the task when there are failed users and when there are not, and the course run grading status is set to complete.

### CODE REVIEW

File: grades/management/commands/check_final_grade_freeze_status.py [hunk 1]
```diff
@@ -74,12 +73,12 @@ def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
                     'Final grades for course "{0}" are not being processed yet'.format(edx_course_key)
                 )
             )
+        message_detail = ', where {0} failed authentication'.format(failed_users_count) if failed_users_count else ""
```
Comment (giocalitri) [inline grades/management/commands/check_final_grade_freeze_status.py]: (very minor) nit: use the same quotes for the 2 strings

File: grades/management/commands/check_final_grade_freeze_status.py [hunk 2]
```diff
@@ -74,12 +73,12 @@ def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
...
         self.stdout.write(
             self.style.SUCCESS(
-                'The students with a final grade are {0}/{1}, where {2} failed authentication'.format(
+                'The students with a final grade are {0}/{1}'.format(
                     FinalGrade.objects.filter(course_run=run).count(),
-                    CachedEnrollment.objects.filter(course_run=run).count(),
-                    failed_users_count
-                )
+                    CachedEnrollment.objects.filter(course_run=run).count()
+                ) + message_detail
```
Comment (giocalitri) [inline grades/management/commands/check_final_grade_freeze_status.py]: please do not use the `+` for joining the two strings

**Comment by giocalitri** [comment]
Also, please add a test to verify that after the course is set to complete, there is no entry in Redis

**Comment by annagav** [comment]
@giocalitri Addressed all comments

**Pull request merged**